### PR TITLE
[Merged by Bors] - feat: port CategoryTheory.Quotient

### PR DESCRIPTION
**Pull request by mpenciak**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
currently there are errors in three parts:

- [ ] `instance category`
- [ ] `functor_map_eq_iff`
- [ ] `lift_unique`

**Comment by bottine** [comment]
I checked the names and things seem in order now. Two changes that I'd like confirmation on:

* There was a commented-out `deriving …` line with an instance later on and a porting note explaining that the `deriving …` line didn't work. I just removed that commented-out `deriving` line, since the porting note is sufficiently informative I believe.
* In the following definition of HomRel, the `isEquiv` constructor was in UpperCamelCase, but I thought it matched items one and five of the [porting wiki](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki) so I changed it to lowerCamel, and now I'm thinking it should maybe just be `is_equiv`? it's confusing.
 
```
/-- A `HomRel` is a congruence when it's an equivalence on every hom-set, and it can be composed
from left and right. -/
class Congruence : Prop where
  /-- `r` is an equivalence on every hom-set. -/
  isEquiv : ∀ {X Y}, IsEquiv _ (@r X Y)
  /-- Precomposition with an arrow respects `r`. -/
  compLeft : ∀ {X Y Z} (f : X ⟶ Y) {g g' : Y ⟶ Z}, r g g' → r (f ≫ g) (f ≫ g')
  /-- Postcomposition with an arrow respects `r`. -/
  compRight : ∀ {X Y Z} {f f' : X ⟶ Y} (g : Y ⟶ Z), r f f' → r (f ≫ g) (f' ≫ g)
#align category_theory.congruence CategoryTheory.Congruence
```

**Comment by jcommelin** [comment]
In general equalities of functors can be quite taxing on the kernel. And mathematically it isn't a very well-behaved notion anyways. That's why we often only state isomorphisms.

File: Mathlib/CategoryTheory/Quotient.lean [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+namespace CategoryTheory
+
+variable {C : Type _} [Category C] (r : HomRel C)
+
+/-- A `HomRel` is a congruence when it's an equivalence on every hom-set, and it can be composed
+from left and right. -/
+class Congruence : Prop where
+  /-- `r` is an equivalence on every hom-set. -/
+  isEquiv : ∀ {X Y}, IsEquiv _ (@r X Y)
```
Comment (jcommelin) [inline Mathlib/CategoryTheory/Quotient.lean:43]: `isEquiv` looks good to me.

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**